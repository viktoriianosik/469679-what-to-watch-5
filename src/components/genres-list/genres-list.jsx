import React from "react";
import PropTypes from "prop-types";
import MoviePropTypes from "../movie/movie-props";
import {connect} from "react-redux";

const GenresList = (props) => {
  const {movies, onGenreClick, activeGenre} = props;
  const genres = [`All genres`];
  let uniqueGenres = [];

  movies.forEach((movie) => genres.push(movie.genre));
  uniqueGenres = [...new Set(genres)].slice(0, 9);

  return (
    <ul className="catalog__genres-list">
      {uniqueGenres.map((genre, i) => (
        <li key={`${i}-item`} className={`catalog__genres-item ${activeGenre === genre && `catalog__genres-item--active`}`}>
          <a
            href="#"
            className="catalog__genres-link"
            onClick={(evt) => {
              evt.preventDefault();
              onGenreClick(genre, movies);
            }}
          >{genre}</a>
        </li>
      ))}
    </ul>
  );
};

GenresList.propTypes = {
  movies: PropTypes.arrayOf(MoviePropTypes).isRequired,
  onGenreClick: PropTypes.func.isRequired,
  activeGenre: PropTypes.string.isRequired,
};

const mapStateToProps = ({PROCESS}) => ({
  activeGenre: PROCESS.genre,
});

export {GenresList};
export default connect(mapStateToProps)(GenresList);
