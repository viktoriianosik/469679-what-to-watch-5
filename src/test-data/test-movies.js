export const testMovies = [
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 1,
    isFavorite: false,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 1.5,
    released: 1994,
    runTime: 153,
    scoresCount: 1635992,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  },
  {
    backgroundColor: `#BDAD8F`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/No_Country_for_Old_Men.jpg`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    director: `Ethan Coen`,
    genre: `Crime`,
    id: 14,
    isFavorite: false,
    name: `No Country for Old Men`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 4.1,
    released: 2007,
    runTime: 122,
    scoresCount: 764976,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  }
];
