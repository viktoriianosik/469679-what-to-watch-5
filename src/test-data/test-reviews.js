export const testReviews = [
  {
    comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2020-10-09T13:38:44.769Z`,
    id: 1,
    rating: 3.3,
    user: {
      id: 19,
      name: `Christina`
    },
  },
  {
    comment: `I love this movie.`,
    date: `2020-13-09T08:38:44.769Z`,
    id: 1,
    rating: 5,
    user: {
      id: 9,
      name: `Maks`
    },
  },
  {
    comment: `Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2019-30-08T13:38:44.769Z`,
    id: 14,
    rating: 8.3,
    user: {
      id: 3,
      name: `Olga`
    },
  },
];
