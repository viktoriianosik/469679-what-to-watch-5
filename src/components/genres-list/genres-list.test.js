import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list";
import {testMovies} from "../../test-data/test-movies";

it(`Render component GenreList correctly`, () => {
  const tree = renderer.create(
      <GenresList
        movies={testMovies}
        onGenreClick={() => {}}
        activeGenre={`All`}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
