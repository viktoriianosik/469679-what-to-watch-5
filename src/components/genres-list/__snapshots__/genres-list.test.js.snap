// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component GenreList correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item false"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
</ul>
`;
